<?xml version="1.0" ?>
<launch>


  <arg name="topic_real_pose" default="/lwr/ee_pose"/>
  <arg name="topic_real_velocity" default="/lwr/ee_vel"/>

  <arg name="topic_target1_position" default="/tracking/target1_position"/>
  <arg name="topic_target2_position" default="/tracking/target2_position"/>

  <arg name="topic_target1_velocity" default="/tracking/target1_velocity"/>
  <arg name="topic_target2_velocity" default="/tracking/target2_velocity"/>

  <arg name="topic_task1_velocity" default="/Task1/desired_velocity"/>
  <arg name="topic_task2_velocity" default="/Task2/desired_velocity"/>

  <arg name="topic_adapted_velocity" default="/lwr/joint_controllers/passive_ds_command_vel"/>


  <!-- Pass in mocap_config_file:=/path/to/config.yaml to change options. -->
  <arg name="mocap_config_file"
       default="$(find cpr_mocap_tracking)/config/mocap_cpr.yml" />

  <node name="mocap_node"
          pkg="mocap_optitrack"
          type="mocap_node"
          respawn="false"
          output="screen"
          launch-prefix=""
          required="true">
      <rosparam file="$(find ds_motion_generator)/config/mocap/mocap.yml" command="load" />
  </node>


  <group ns="tracking">
    <param name="base_topic_name" value="/mocap/kuka_base/pose"/>
    <param name="object_1_topic_name" value="/mocap/kuka_ee/pose"/>
    <param name="object_2_topic_name" value="/mocap/fake_target/pose"/>
    <param name="target_1_topic_name" value="$(arg topic_target1_position)"/>
    <param name="target_2_topic_name" value="$(arg topic_target2_position)"/>
    <param name="topic_target1_vel" value="$(arg topic_target1_velocity)"/>
    <param name="topic_target2_vel" value="$(arg topic_target2_velocity)"/>
    <param name="filter_factor" value="0.1"/>
    <param name="v_max" value="0.5"/>
    <node name="tracking_node" pkg="ds_motion_generator" type="target_tracker_node" output="screen"/>
  </group>

  <group ns="ds1">
    <param name="input_topic_name" value="$(arg topic_real_pose)"/>
    <param name="target_topic_name" value="$(arg topic_target1_position)"/>
    <param name="output_topic_name" value="$(arg topic_task1_velocity)"/>
    <param name="output_filtered_topic_name" value="/Task1/filter/desired_velocity"/>
    <rosparam command="load" file="$(find ds_motion_generator)/config/simple_ds.yml"/>
    <node name="ds_node" pkg="ds_motion_generator" type="active_motionGenerator_node" output="screen"/>
  </group>

  <group ns="ds2">
    <param name="input_topic_name" value="$(arg topic_real_pose)"/>
    <param name="target_topic_name" value="$(arg topic_target2_position)"/>
    <param name="output_topic_name" value="$(arg topic_task2_velocity)"/>
    <param name="output_filtered_topic_name" value="/Task1/filter/desired_velocity"/>
    <rosparam command="load" file="$(find ds_motion_generator)/config/simple_ds.yml"/>
    <node name="ds_node" pkg="ds_motion_generator" type="active_motionGenerator_node" output="screen"/>
  </group>


  <group ns="handover_adaptation">
    <param name="topic_target1_velocity"   value="$(arg topic_target1_velocity)"/>
    <param name="topic_target2_velocity"   value="$(arg topic_target2_velocity)"/>
    <param name="topic_task1_velocity"     value="$(arg topic_task1_velocity)"/>
    <param name="topic_task2_velocity"     value="$(arg topic_task2_velocity)"/>
    <param name="topic_adapted_velocity"     value="$(arg topic_adapted_velocity)"/>    
    <node name="TA_node" pkg="task_adaptation" type="TwoHandoversAdaptor" output="screen"/>

    <!-- <node name="TA_record" pkg="task_adaptation" type="record_adaptation_node" output="screen"/> -->


  </group>


  <node pkg="rostopic" type="rostopic" name="angle_setting" args="pub /lwr/joint_controllers/passive_ds_command_orient geometry_msgs/Quaternion '{x: 0.0, y: 1.0, z: 0.0, w: 0.0}' -r 100" />

  <node name="rqt_reconfigure" pkg="rqt_reconfigure" type="rqt_reconfigure" />




</launch>
